lib/src/themes/expression/parser/expression_parser.dart

```diff
@@ -96,6 +99,25 @@ class ExpressionParser {
       }
     }
     if (json is Map) {
+      // legacy categorical expressions
+      // https://docs.mapbox.com/style-spec/reference/other
+      if (json['type'] == 'categorical') {
+        final property = json['property'];
```

Comment:
would we get better code reuse by moving this inside the `stops is List` check below?

Reply:
fixed